README.md

```diff
@@ -1,2 +1,161 @@
+
+## Implementation Requirements
+
+We will do an in-class design activity for this project. While you should read through these requirements before that exercise, we do not recommend beginning work until we have finished classroom work.
+
+### Wave 1 - Listing Channels and Users
+
+As a user, when I launch this program... 
+
+- I should see information about how many channels and users were loaded
+- I should then be given three options for how to interact with the program:
+  - List users
+  - List channels
+  - Quit
+
+As a user who is at the program's input prompt...
+
+- When I type "list users", I should see a list of all the users in the Slack workspace. This list should include username, real name, and Slack ID.
+- When I type "list channels", I should see a list of all the channels for that workspace. This list should include the channel's name, topic, member count, and Slack ID.
+- When I type "quit", the program should exit.
+- After completing any command other than "quit", the program should reprint the list of commands and ask for another input.
+
+**Hint:** You may want to investigate the [Table Print gem](https://github.com/arches/table_print) to handle formatting tables.
+
+### Wave 2 - Selecting and Showing Details
+
+**Vocab:** channels and users are both _recipients_.
+
+As a user at the input prompt...
+
+- I should see three new options:
+  - Select user
+  - Select channel
+  - Details
+- When I type "select user", I should be able to supply a username or Slack ID. The corresponding user should be the "selected" recipient.
+- When I type "select channel", I should be able to supply a channel name or Slack ID. The corresponding channel should be the "selected" recipient.
+- For selecting both users and channels, if no user/channel has that name or ID, the program should let me know and return to the main command loop.
+- When I type "details", the program should print out details for the currently selected recipient. What information is printed depends on whether it's a channel or a user.
+  - If no recipient is currently selected, the program should let me know and return to the main command prompt.
+
+### Wave 3 - Sending Messages
+
+As a user at the input prompt...
+
+- I should see one new option: send message.
+- When I type "send message", if a recipient is selected the program should ask me to type out a message, which will be sent to that recipient.
+  - If no recipient is currently selected, the program should let me know and return to the main command prompt.
+
+## Optional Enhancements
+
+Don't even think about reading these until all the primary requirements are complete.
+
+- As a user, I can change global settings for the program
+  - These settings include the username for the bot and an emoji to use as an icon
+  - When I change these settings, the program should save them in the JSON format in a file named `bot-settings.json`. When I restart the program, it should reload those settings.
```

Comment:
by this point, have we given them a good resource to think about reading a json file? I'm assuming this optional is really "above and beyond," so I want to confirm.

Reply:
We have not. The intent here is to give something really meaty to work on.